src/createElement.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * 일반 태그이거나 함수형 컴포넌트면 createElement 함수 실행
+ */
+export const createElement = (type: string | Function, props: JSX.Props, ...children: any[]) => {
+  // 함수형 컴포넌트면 props랑 전개한 children 내려줘서 재귀적으로 createElement 실행
+  if (typeof type === 'function') {
+    return type(props, ...children);
+  }
+
+  return {
+    type,
+    props: {
+      ...props,
+      children:
+        children.length === 1 // children이 하나만 있으면 배열 말고 객체로 처리
+          ? validTextNode(children[0])
+          : children.map((child) => validTextNode(child)), // children이 여러개면 배열로 묶어서 처리
+    },
+  };
+};
+
+/**
+ * leaf 노드에 있는 텍스트 노드면 type: createTextNode로 처리
+ */
+const validTextNode = (child: any) => {
```

Comment:
```suggestion
const processTextNode = (child: any) => { // 텍스트 노드를 처리하는 함수
```
개인적인 의견으로는 `validTextNode` 함수 이름이 텍스트 노드로 처리될 수 있는지 유효성 검사만 거칠 것 같은데 
실제로는 `createTextNode` 함수를 통해 텍스트 노드로 변환하는 과정이 포함되어 있는 것 같아서 실제로 텍스트 노드를 변환하고 처리하는 과정을 알릴 수 있는 함수명도 좋을 것 같다는 생각이 드는 것 같네요🤔